src/V4/PeanutRouter.sol

```diff
@@ -0,0 +1,120 @@
+contract PeanutV4Router is Ownable {
+    using SafeERC20 for IERC20;
+
+    PeanutV4 public peanut;
+    address public squidAddress;
+
+    constructor(address _peanutAddress, address _squidAddress) {
+        peanut = PeanutV4(_peanutAddress);
+        squidAddress = _squidAddress;
+    }
+
+     /**
+     * @notice Function to withdraw a peanut deposit to a different chain.
+     * @dev We could include something like "expiry" in this function to prevent X-chain
+     * transfers getting stuck in Axelar (Squid works on top of Axelar) in case gas price
+     * on the destination chain increases while the tx is pending in the mempool on the source chain.
+     * But it doesn't really matter since all transfers are paid by us and we can always increase
+     * the axelar fee (e.g. from our backend server).
+     * @param _depositIndex The index of the deposit in peanut.
+     * @param _withdrawalSignature signature to withdraw from peanut.
+     * @param _squidFee squid router fee.
+     * @param _peanutFee fee amount taken by peanut (this contract) for routing.
+     * @param _squidData calldata fro the squid router
+     * @param _routingSignature signed _squidFee, _peanutFee and _squidData
+     */
+    function withdrawAndBridge(
+        uint256 _depositIndex,
+        bytes calldata _withdrawalSignature,
+        uint256 _squidFee,
+        uint256 _peanutFee,
+        bytes calldata _squidData,
+        bytes calldata _routingSignature
+    ) public payable {
+        PeanutV4.Deposit memory deposit = peanut.getDeposit(_depositIndex);
+
+        // We must first validate _routingSignature to prevent front-running
+        // The signature structure follows version 0x00 from EIP-191
+        bytes32 digest = keccak256(
+            abi.encodePacked(
+                bytes2(0x0019),
+                address(this),
+                block.chainid,
+                address(peanut),
+                _depositIndex,
+                squidAddress,
+                _squidFee,
+                _peanutFee,
+                _squidData
+            )
+        );
+        address routingSigner = ECDSA.recover(digest, _routingSignature);
+        require(routingSigner == deposit.pubKey20, "WRONG ROUTING SIGNER");
+
+        require(_squidFee == msg.value, "msg.value MUST BE THE SQUID FEE");
+        require(deposit.contractType == 0 || deposit.contractType == 1, "UNSUPPORTED contractType");
+        require(_peanutFee < deposit.amount, "TOO HIGH FEE");
+
+        peanut.withdrawDepositAsRecipient(_depositIndex, address(this), _withdrawalSignature);
+
+        uint256 amountToBridge = deposit.amount - _peanutFee;
+        uint256 ethAmountToSquid = msg.value;
+        if (deposit.contractType == 0) { // ETH deposit
+            ethAmountToSquid += amountToBridge;
+        } else if (deposit.contractType == 1) { // ERC20 deposit
+            IERC20 token = IERC20(deposit.tokenAddress);
+            token.safeApprove(address(squidAddress), amountToBridge);
```

Comment:
> Why not use safeIncreaseAllowance?

Doesn't make much a difference I think.

I went with `approve` since with `increaseAllowance` it's easier to mismanage something and end up having an extra approval for the router. With `approve` we are always sure the allowance is never higher than the amount that is being transferred in this particular call.

But doing `increaseAllowance` also makes sense since every transaction will be more atomic. So I personally don't have a preference here, and so let me then switch to `increaseAllowance` if you prefer this one.